biorxiv/polka_subset_experiment/03_rerun_analyses_with_added_subset.py

```diff
@@ -0,0 +1,603 @@
+# -
+
+# ### Document version count plot
+
+biorxiv_published_distances = pd.read_csv(
+    "../publication_delay_experiment/output/preprint_published_distances.tsv", sep="\t"
+)
+biorxiv_published_distances["time_to_published"] = pd.to_timedelta(
+    biorxiv_published_distances["time_to_published"]
+)
+biorxiv_published_distances["days_to_published"] = biorxiv_published_distances[
+    "time_to_published"
+].dt.days
+biorxiv_published_distances.head()
+
+# +
+# Get smoothed linear regression line
+x = biorxiv_published_distances.version_count.values.tolist()
+
+y = biorxiv_published_distances.time_to_published.apply(
+    lambda x: x / timedelta(days=1)
+).tolist()
+
+xseq_2 = np.linspace(np.min(x), np.max(x), 80)
+
+results_2 = linregress(x, y)
+print(results_2)
+# -
+
+x_line = np.array(
+    [
+        biorxiv_published_distances["version_count"].min(),
+        biorxiv_published_distances["version_count"].max(),
+    ]
+)
+y_line = x_line * results_2.slope + results_2.intercept
+
+# Graph here?
+plt.figure(figsize=(8, 5))
+g = sns.violinplot(
+    x="version_count",
+    y="days_to_published",
+    data=biorxiv_published_distances,
+    cut=0,
+    scale="width",
+    palette="YlGnBu",
+)
+_ = g.set_ylabel("Time Elapsed Until Preprint is Published (Days)")
+_ = g.set_xlabel("# of Preprint Versions")
+_ = g.plot(x_line - 1, y_line, "--k")
+_ = g.scatter(
+    polka_published_preprint_df["version_count"] - 1,
+    polka_published_preprint_df["days_to_published"],
+    c="red",
+    s=12,
+)
+_ = g.annotate(f"Y={results_2.slope:.2f}*X+{results_2.intercept:.2f}", (7, 1470))
+_ = g.set_xlim(-0.5, 11.5)
+_ = g.set_ylim(0, g.get_ylim()[1])
+plt.savefig("output/figures/version_count_vs_publication_time_violin.svg", dpi=500)
+plt.savefig("output/figures/version_count_vs_publication_time_violin.png", dpi=500)
+
+# ### Document embedding pair
+
+polka_published_df = polka_published_df.set_index("document")
+polka_published_df.head()
+
+polka_preprints_df = polka_preprints_df.set_index("biorxiv_base").drop(
+    ["document", "version_count"], axis=1
+)
+polka_preprints_df.head()
+
+dist = np.diag(
+    cdist(
+        polka_preprints_df.loc[polka_published_preprint_df["biorxiv_base"]],
+        polka_published_df.loc[polka_published_preprint_df["PMCID"]],
+    )
+)
+print(dist)
+
+polka_published_preprint_df = polka_published_preprint_df.assign(doc_distances=dist)
+polka_published_preprint_df.head()
+
+# +
+# Get smoothed linear regression line
+x = biorxiv_published_distances.doc_distances.values.tolist()
+
+y = biorxiv_published_distances.time_to_published.apply(
+    lambda x: x / timedelta(days=1)
+).tolist()
+
+xseq_2 = np.linspace(np.min(x), np.max(x), 80)
+
+results_2 = linregress(x, y)
+print(results_2)
+# -
+
+x_line = np.array(
+    [
+        biorxiv_published_distances["doc_distances"].min(),
+        biorxiv_published_distances["doc_distances"].max(),
+    ]
+)
+y_line = x_line * results_2.slope + results_2.intercept
+
+# graph here?
```

Comment:
I think this is saying the same thing as  above, that the longer it takes to publish, the more changes your preprint will undergo